### PR TITLE
Update CONTRIBUTING.md to add line length limit

### PR DESCRIPTION
**Pull request by jakebolewski**
Prompted by encountering some 250+ character lines in the linear algebra library

Also add naming conventions for modules, types, and methods.

[ci skip]

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -95,14 +95,17 @@ Build as usual, and do `make clean testall` to test your contribution. If your c
 
 ##### General Formatting Guidelines For Julia code contributions
 
- - 4 space indent
+ - 4 spaces per indentation level, no tabs
  - use whitespace to make the code more readable
- - no whitespace at the end of a line
+ - no whitespace at the end of a line (trailing whitespace)
  - comments are good, especially when they explain the algorithm
+ - try to adhere to a 92 character line length limit
```
Comment (IainNZ) [inline CONTRIBUTING.md]: Where did 92 come from?

Comment (jakebolewski) [inline CONTRIBUTING.md]: An arbitrary number less than 250.

Comment (ViralBShah) [inline CONTRIBUTING.md]: Sounds like Fortran!

Comment (jiahao) [inline CONTRIBUTING.md]: Hard to beat the classics:

latest

[ref](http://futurama.wikia.com/wiki/Olde_Fortran)

**Comment by iamed2** [comment]
Python's PEP8 suggests 99 as a hard max, why not that? Seems like as good a reason as any.

File: CONTRIBUTING.md (hunk 1 above)
Comment (jakebolewski) [inline CONTRIBUTING.md]: with optional command line flag, you get to upgrade to 132 characters per line :-)

**Comment by elextr** [comment]
`rand()*250` gives 100.00978344188333 (in Julia 0.3.5) so 99 looks like it :wink:

**Comment by iamed2** [comment]
Oh weird, it gives me 134.59251205400307 on 0.4. Is this a regression? ;)

**Comment by elextr** [comment]
On 21 January 2015 at 15:12, iamed2 <notifications@github.com> wrote:

> Oh weird, it gives me 134.59251205400307 on 0.4. Is this a regression? ;)
>

​No, it is random :)

Cheers
Lex

PS today I got ​169.56878644092737, I presume it depends on your systems
entropy, it was just lucky that time :)

**Comment by eschnett** [comment]
julia> rand()*250
80.38026797235948

And I only needed two tries to get that result!

-erik

> On Jan 21, 2015, at 2:14 , elextr <notifications@github.com> wrote:
> 
> On 21 January 2015 at 15:12, iamed2 <notifications@github.com> wrote:
> 
> > Oh weird, it gives me 134.59251205400307 on 0.4. Is this a regression? ;)
> >
> 
> ​No, it is random :)
> 
> Cheers
> Lex
> 
> PS today I got ​169.56878644092737, I presume it depends on your systems
> entropy, it was just lucky that time :)
> 
> 
> 
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/JuliaLang/julia/pull/9848#issuecomment-70785603>.
> >
> —
> Reply to this email directly or view it on GitHub.
> 

--
Erik Schnetter <schnetter@gmail.com>
http://www.perimeterinstitute.ca/personal/eschnetter/

My email is as private as my paper mail. I therefore support encrypting
and signing email messages. Get my PGP key from https://sks-keyservers.net.

**Pull request merged**